### PR TITLE
Clear search text input in new group page after select a user

### PR DESCRIPTION
**Pull request by K4tsuki**
### Details

Clear search input in New Group page after select a user.
### Fixed Issues

$ https://github.com/Expensify/App/issues/6872

### Tests

1. Open New Group page
2. Type some character to search
3. Select a user
4. After select a user, verify the search text box is cleared.
- [x] Verify that no errors appear in the JS console

### QA Steps

- [X] Verify that no errors appear in the JS console
1. Open New Group page
2. Type some character to search
3. Select a user
4. After select a user, verify the search text box is cleared.

### Tested On

- [X] Web
- [X] Mobile Web
- [ ] Desktop
- [ ] iOS
- [X] Android

#### Web

https://user-images.githubusercontent.com/1313124/152484169-982d39c5-9435-4871-b88f-d38021f0c20e.mp4

#### Mobile Web

https://user-images.githubusercontent.com/1313124/152483905-aee95aa3-6b52-40ca-b4f5-18a22ef83d17.mp4

#### Android

https://user-images.githubusercontent.com/1313124/152492822-9859b593-9262-41d0-886f-873df76024ad.mp4

### CODE REVIEW

**Review by parasharrajat — DISMISSED** [review]
LGTM. For now, I am skipping the screenshot check for Desktop and IOS which should work as change is purely logical. 

cc: @pecanoro 

:ribbon: :eyes: :ribbon:  C+ reviewed

File: src/components/TextInput/BaseTextInput.js [hunk 1]
```diff
@@ -53,6 +53,9 @@ class BaseTextInput extends Component {
         }
 
         this.value = this.props.value;
+        if (this.props.value === '') {
```
Comment (pecanoro) [inline src/components/TextInput/BaseTextInput.js:56]: It looks good but can you leave a comment before the if condition explaining why we need to do this?

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: src/components/TextInput/BaseTextInput.js (hunk 1 above)
Comment (K4tsuki) [reply src/components/TextInput/BaseTextInput.js:60]: I have add the comment

File: src/components/TextInput/BaseTextInput.js [hunk 2]
```diff
@@ -54,6 +54,14 @@ class BaseTextInput extends Component {
 
         this.value = this.props.value;
 
+        // Since BaseTextInput value is depend on this.value
+        // Providing value to BaseTextInput via props.value will be late one step behind
+        // So at componentDidUpdate we update the real value of TextInput
+        // If this.props.value is empty string, we must clear the TextInput
```
Comment (parasharrajat) [inline src/components/TextInput/BaseTextInput.js:60]:
```suggestion
        // In some cases, When the value prop is empty, it is not properly updated on the TextInput due to its uncontrolled nature, thus manually clearing the TextInput. 
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Review by pecanoro — DISMISSED** [review]

**Review by parasharrajat — CHANGES_REQUESTED** [review]
lint issues @K4tsuki....

**Comment by K4tsuki** [comment]
lint error fixed

**Review by pecanoro — APPROVED** [review]

**Pull request merged**